tests/test_indexer.py

```diff
@@ -115,14 +116,19 @@ def tearDown(self):
         self.storageHelper = None
 
     def test_process_new_indexable_object(self):
-        bundle_key = self.load_test_data_bundle_for_path("fixtures/smartseq2/paired_ends")
-        sample_event = self.create_sample_bundle_created_event(bundle_key)
-        self.process_new_indexable_object(sample_event, logger)
+        bundle_key = self.insert_and_process_new_indexable_object()
         search_results = self.get_search_results(smartseq2_paired_ends_query, 1)
         self.assertEqual(1, len(search_results))
         self.verify_index_document_structure_and_content(search_results[0], bundle_key,
                                                          files=smartseq2_paried_ends_indexed_file_list)
 
+    def test_process_removed_object(self):
+        bundle_key = self.insert_and_process_new_indexable_object()
+        sample_event = self.create_sample_bundle_removed_event(bundle_key)
+        self.process_removed_object(sample_event, logger)
+        search_results = self.get_search_results(smartseq2_paired_ends_query, 0)
```

Comment:
The search that you are performing to verify that the document is no longer there should also be performed before removal to ensure that the document is there and is found by the search. The document being loaded sometimes changes, and it could change in a way that the search no longer matches it, and in that case, as the test stands now, de-indexing could fail but the test would still pass. Let me know if you have questions or need clarification regarding this case.